FHIR/FSHOnline
# Integrate FPL v2

~_Note: This PR is in marked as a draft because the `FSHHelpers.test.jss` suite does not run yet but the general approach to integrating FPL v2 and new SUSHI and GoFSH is stable._~ **Jan 7 update:** this is ready for review now.

**Description:**
This PR updates FSH Online to use the new FPL v2 and newer versions of SUSHI and GoFSH that use FPL v2 as well.

This PR cleans up a good amount of code that was moved into FPL. I also tried to use some newer SUSHI functions where possible, rather than having FSH Online write its own versions.

The way I had to create to make some temporary definitions for GoFSH is a little odd, but it was the only way I could figure out how to use a FHIRProcessor to `processConfig` to get all the dependencies from an IG resource, which we needed to properly initialize the BrowserBasedPackageCache.

Note the following changes still need to be made to this branch before it can be merged:

- [x] The `FSHHelpers.test.js` file cannot load the `sql-wasm.wasm` file correctly right now, so that needs to be figured out. The `assetsInclude` configuration field is what makes this work when running a development or production server, so I guess the testing equivalent needs to be figured out.
- [x] Once the FSHHelpers.test.js file properly loads, the tests need to be updated to mock out the pieces that FPL is doing now, because it still is mocking out parts from the old FSH Online Processing functions, which do not exist anymore.
- [x] The correct SUSHI and GoFSH versions need to be used in the `package.json` once they're available and the versions need to be updated in the TopBar component.

**Jan 7 update: these have all been done.**

**Testing Instructions:**
Ttry using this locally in your browser to make sure there isn't functionality that I've missed that needs to be added in FPL or here. You should be able to:

- download and cache packages in your browser's IndexedDB -- you can clear your IndexedDB to ensure new packages are being downloaded, but also the default name of the database has changed, so the first time you run this, it should try to download and cache the package
- load packages from the cache if they're already downloaded
- run SUSHI without configured dependencies and with one or more configured dependencies
- run GoFSH without configured dependencies, with one or more configured dependencies, and with dependencies that are only specified in an ImplementationGuide resources

**Related Issue:**
N/A
